### ISSUE TITLE
Snowflake dialect implementation issue

### ISSUE THREAD

**Issue by AlessandroLollo** [sodadata/soda-sql#65]
Hi guys,

I've tried using SodaSQL with Snowflake, but I've had an issue with `soda init`.

First, I've run `soda create .`, which created the default `warehouse.yml`.
I modified it to look like the following:
```
name: .
connection:
  type: snowflake
  username: MYUSERNAME
  password: env_var(SNOWFLAKE_PWD)
  account: MYACCOUNTNAME
  database: MYDB
  warehouse: MYWH
  schema: MYSCHEMA
```

In `~/.soda/env_vars.yml` I have:
```
.:
  SNOWFLAKE_PWD: '<my secret password>'
```

When I run `soda init .` I get the following error:
```
/Users/alessandro.lollo/.pyenv/versions/3.8.3/envs/playground/lib/python3.8/site-packages/pandas/compat/__init__.py:120: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError.
  warnings.warn(msg)
/Users/alessandro.lollo/.pyenv/versions/3.8.3/envs/playground/lib/python3.8/site-packages/snowflake/connector/options.py:78: UserWarning: You have an incompatible version of 'pyarrow' installed (2.0.0), please install a version that adheres to: 'pyarrow<0.18.0,>=0.17.0; extra == "pandas"'
  warn_incompatible_dep('pyarrow', _installed_pyarrow_version.version, _expected_pyarrow_version)
  | Querying warehouse for tables
  | Exception: TODO override and implement this abstract method
Traceback (most recent call last):
  File "/Users/alessandro.lollo/.pyenv/versions/3.8.3/envs/playground/lib/python3.8/site-packages/sodasql/cli/cli.py", line 196, in init
    warehouse.dialect.sql_tables_metadata_query())
  File "/Users/alessandro.lollo/.pyenv/versions/3.8.3/envs/playground/lib/python3.8/site-packages/sodasql/scan/dialect.py", line 98, in sql_tables_metadata_query
    raise RuntimeError('TODO override and implement this abstract method')
RuntimeError: TODO override and implement this abstract method
```

A part from `pyarrow` and `lzma` warnings, it seems that the `sql_tables_metadata_query` method is not implemented for Snowflake. I have not looked into SodaSQL code, though.

**Comment by milanaleksic** [comment]
Hi @AlessandroLollo , thank you for for your information. 

I see from the stack trace that you are using `pyenv` that handles multiple python installations, but is not a perfect isolation tool. Did you try creating a standard venv and then installing the dependencies of the project, and only them? That way chance of collision is lower

```
mkdir sodasqlcli
cd sodasqlcli
python3 -m venv .venv
source .venv/bin/activate
pip install pip
# now follow the steps of the 5 minute tutorial, like `pip install soda-sql`
```

**Comment by AlessandroLollo** [comment]
Hi @milanaleksic,
thanks for your suggestion!

With a brand new virtualenv and `sodasql==2.0.0b2` I get the same error mentioned in my previous comment.
With latest version, namely `sodasql==2.0.0b3` I get the following error message

```
Traceback (most recent call last):
  File "/Users/alessandro.lollo/.pyenv/versions/sodasql/bin/soda", line 6, in <module>
    from sodasql.cli.cli import main
  File "/Users/alessandro.lollo/.pyenv/versions/3.8.3/envs/sodasql/lib/python3.8/site-packages/sodasql/cli/cli.py", line 19, in <module>
    from sodasql.cli.scan_initializer import ScanInitializer
  File "/Users/alessandro.lollo/.pyenv/versions/3.8.3/envs/sodasql/lib/python3.8/site-packages/sodasql/cli/scan_initializer.py", line 19, in <module>
    from sodasql.scan.warehouse import Warehouse
  File "/Users/alessandro.lollo/.pyenv/versions/3.8.3/envs/sodasql/lib/python3.8/site-packages/sodasql/scan/warehouse.py", line 15, in <module>
    from sodasql.scan.warehouse_yml import WarehouseYml
  File "/Users/alessandro.lollo/.pyenv/versions/3.8.3/envs/sodasql/lib/python3.8/site-packages/sodasql/scan/warehouse_yml.py", line 15, in <module>
    from sodasql.soda_server_client.soda_server_client import SodaServerClient
ModuleNotFoundError: No module named 'sodasql.soda_server_client'
```

I'm running `sodasql init .` from within the directory where `warehouse.yml` is stored

**Comment by milanaleksic** [comment]
The new beta is incoming very soon (minutes) which should fix that issue (#1766); apologies for the inconvenience.

**Comment by AlessandroLollo** [comment]
Hi @milanaleksic,

no worries!
I'm happy to contribute and report any issues I find 🙂

**Comment by milanaleksic** [comment]
@AlessandroLollo could you retry? Just do a `pip install soda-sql --upgrade` to get the newest 2.0.0b4

**Comment by AlessandroLollo** [comment]
With `sodasql==2.0.0b4` I get the same error reported in my first comment.

**Comment by milanaleksic** [comment]
Thanks, we'll take a look

**Comment by AlessandroLollo** [comment]
> Thanks, we'll take a look

Please let me know if there's anything else I can do to help you guys figure it out

**Comment by tombaeyens** [comment]
I ll check what I can find...

**Comment by tombaeyens** [comment]
@AlessandroLollo I just released 2.0.0b5 Can you give it another try?

Though this was not a problem for you, I've also updated the installation docs.  While fixing it we encountered some issues with versions lower than 21 of `pip`.  https://docs.soda.io/soda-sql/getting-started/installation.html

**Comment by AlessandroLollo** [comment]
> @AlessandroLollo I just released 2.0.0b5 Can you give it another try?
> 
> Though this was not a problem for you, I've also updated the installation docs. While fixing it we encountered some issues with versions lower than 21 of `pip`. https://docs.soda.io/soda-sql/getting-started/installation.html

Hey @tombaeyens, now it works as expected! 👍
